util/design/secded_gen.py

```diff
@@ -317,6 +322,52 @@ def verify(cfgs):
     return error
 
 
+def ecc_encode(codetype: str, k: int, dataword: int) -> Tuple[int, int]:
```

Comment:
@eunchan i added the types here because i was personally getting confused between the hex string, binary string etc.
But is there any way to actually make use of this type assignment without running some kind of pylint..?

Reply:
typing is kind of useless in the current python as far as I know. This is for static typing to check any suspicious code in build time. But Python does not have build time at all right? :) 

So, yeah. pylint would give hint that possible errors.